radio/src/VirtualFS.h

```diff
@@ -0,0 +1,311 @@
+  {
+    if( _instance == nullptr)
+      _instance = new VirtualFS();
+    return _instance;
+  }
+
+  bool format();
+  const char * checkAndCreateDirectory(const char * path);
+  bool isFileAvailable(const char * path, bool exclDir);
+  bool isFilePatternAvailable(const char * path, const char * file, const char * pattern = nullptr, bool exclDir = true, char * match = nullptr);
+  char* getFileIndex(char * filename, unsigned int & value);
+
+  const std::string& getCurWorkDir()  { return curWorkDir;}
+  int changeDirectory(const std::string& path);
+  int openDirectory(VfsDir& dir, const char * path);
+  int readDirectory(VfsDir& dir, VfsFileInfo& info, bool firstTime = false);
+  int closeDirectory(VfsDir& dir);
+
+  int rename(const char* oldPath, const char* newPath);
+
+  uint32_t flashGetNoSectors();
+  uint32_t flashGetSize();
+  uint32_t flashGetFreeSectors();
+
+  //#if !defined(BOOT)
+  //inline const char * SDCARD_ERROR(FRESULT result)
+  //{
+  //  if (result == FR_NOT_READY)
+  //    return STR_NO_SDCARD;
+  //  else
+  //    return STR_SDCARD_ERROR;
+  //}
+  //#endif
+
+  // NOTE: 'size' must = 0 or be a valid character position within 'filename' array -- it is NOT validated
+  const char * flashGetBasename(const char * path);
+
+  #if defined(PCBX12S)
+    #define OTX_FOURCC 0x3478746F // otx for X12S
+  #elif defined(RADIO_T16)
+    #define OTX_FOURCC 0x3F78746F // otx for Jumper T16
+  #elif defined(RADIO_T18)
+    #define OTX_FOURCC 0x4078746F // otx for Jumper T18
+  #elif defined(RADIO_TX16S)
+    #define OTX_FOURCC 0x3878746F // otx for Radiomaster TX16S
+  #elif defined(PCBX10)
+    #define OTX_FOURCC 0x3778746F // otx for X10
+  #elif defined(PCBX9E)
+    #define OTX_FOURCC 0x3578746F // otx for Taranis X9E
+  #elif defined(PCBXLITES)
+    #define OTX_FOURCC 0x3B78746F // otx for Taranis X-Lite S
+  #elif defined(PCBXLITE)
+    #define OTX_FOURCC 0x3978746F // otx for Taranis X-Lite
+  #elif defined(RADIO_T12)
+    #define OTX_FOURCC 0x3D78746F // otx for Jumper T12
+  #elif defined(RADIO_TLITE)
+    #define OTX_FOURCC 0x4278746F // otx for Jumper TLite
+  #elif defined(RADIO_TPRO)
+    #define OTX_FOURCC 0x4678746F // otx for Jumper TPro
+  #elif defined(RADIO_TX12)
+    #define OTX_FOURCC 0x4178746F // otx for Radiomaster TX12
+  #elif defined(RADIO_ZORRO)
+    #define OTX_FOURCC 0x4778746F // otx for Radiomaster Zorro
+  #elif defined(RADIO_T8)
+    #define OTX_FOURCC 0x4378746F // otx for Radiomaster T8
+  #elif defined(PCBX7)
+    #define OTX_FOURCC 0x3678746F // otx for Taranis X7 / X7S / X7 Express / X7S Express
+  #elif defined(PCBX9LITES)
+    #define OTX_FOURCC 0x3E78746F // otx for Taranis X9-Lite S
+  #elif defined(PCBX9LITE)
+    #define OTX_FOURCC 0x3C78746F // otx for Taranis X9-Lite
+  #elif defined(PCBX9D) || defined(PCBX9DP)
+    #define OTX_FOURCC 0x3378746F // otx for Taranis X9D
+  #elif defined(PCBNV14)
+    #define OTX_FOURCC 0x3A78746F // otx for NV14
+  #elif defined(PCBSKY9X)
```

Comment:
PL18 0x4878746F
https://github.com/EdgeTX/edgetx/blob/92dac552f4de61441c2279063eed17bda8f64f68/radio/src/sdcard.h#L173-L174

Reply:
The defined are just left over from using sdcard.h as a base.
The path definitions at the beginning of the file will stay, I thjink, an be removed from sdcard.h. But then can only happen when everysthing is moved over to the VirtualFS.

Reply:
Do we actually still need these defs now?

Reply:
I think so, but not here, but only in sdcard.h

Reply:
I believe this is used only for conversions from binary format.

Reply:
What is the plan regarding binary format? Require 2.6 as a conversion point for future users or do you plan to keep compatibility in 2.7 and later?

Reply:
Either 2.7 or 2.8 will probably drop binary conversions. We shall see what the community thinks.